src/graph.rs

```diff
@@ -373,3 +406,35 @@ impl From<GraphNodeRef> for Value {
         Value::GraphNode(value)
     }
 }
+
+impl fmt::Display for GraphNodeRef {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "[graph node {}]", self.0)
+    }
+}
+
+impl DisplayWithGraph for GraphNodeRef {
+    fn fmt(&self, f: &mut fmt::Formatter, _graph: &Graph) -> fmt::Result {
+        write!(f, "[graph node {}]", self.0)
+    }
+}
+
+/// Trait to Display with a given Context
+pub trait DisplayWithGraph
+where
+    Self: Sized,
+{
+    fn fmt<'tree>(&self, f: &mut fmt::Formatter, graph: &Graph<'tree>) -> fmt::Result;
+
+    fn display_with<'a, 'tree>(&'a self, graph: &'a Graph<'tree>) -> Box<dyn fmt::Display + 'a> {
+        struct Impl<'a, 'tree, T: DisplayWithGraph>(&'a T, &'a Graph<'tree>);
+
+        impl<'a, 'tree, T: DisplayWithGraph> fmt::Display for Impl<'a, 'tree, T> {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                self.0.fmt(f, self.1)
+            }
+        }
+
+        Box::new(Impl(self, graph))
+    }
```

Comment:
Normally I'd suggest that we find a way to avoid the `Box` here, since that means a heap allocation.  If this were a standalone function, and not inside of a trait, you could have the return type be `impl fmt::Display + 'a`, which would avoid the allocation.  But I'm guessing you want to keep this as a method in the trait so that you can call it as `value.display_with` instead of `display_with(value`.

So I see two options:

1. Keep it as `Box`, because this isn't part of the hot path of anything.

2. Move `display_with` outside of the trait, even though it changes how you'd call it.  (That's what I did with the equivalent over in [`stack-graphs`](https://github.com/github/stack-graphs/blob/8f95e1a66db7a1ae9272ba71da1ff33781995f78/src/paths.rs#L78-L92)).